File: app/src/main/java/ua/com/todd/smsforwadinga/smsforwading/MailSenderService.java
package ua.com.todd.smsforwadinga.smsforwading;

import android.app.IntentService;
import android.content.Intent;

import org.apache.commons.mail.DefaultAuthenticator;
import org.apache.commons.mail.Email;
import org.apache.commons.mail.SimpleEmail;

import java.util.List;

import ua.com.todd.baseapp.utils.Utils;
import ua.com.todd.smsforwadinga.smsforwading.data.HelperFactory;
import ua.com.todd.smsforwadinga.smsforwading.managers.PreferenceManager;

public class MailSenderService extends IntentService {

    public MailSenderService() {
        super(MailSenderService.class.getName());
    }

    private String login;
    private String pass;
    private String host;
    private int port;

    @Override
    public void onCreate() {
        super.onCreate();
        MyApplication app = (MyApplication) getApplication();
        PreferenceManager preferenceManager = app.getPreferenceManager();
        login = preferenceManager.restorePassword();
        pass = preferenceManager.restorePassword();
        host = preferenceManager.restoreHost();
        port = preferenceManager.restorePort();
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        if (Utils.isNotNull(login, pass, host, port))
            try {
                List<Sms> smsList = HelperFactory.getHelper().getSmsDAO().getAllSms();
                Email email = new SimpleEmail();
                email.setHostName(host);
                email.setSmtpPort(port);
                email.setStartTLSEnabled(true);
                email.setAuthenticator(new DefaultAuthenticator(login, pass));
                email.setFrom(login);
                email.setSubject("TestMail");
                email.setMsg(smsList.toString());
                email.addTo("timandriyaschenko@gmail.com");
                email.send();
                HelperFactory.getHelper().getSmsDAO().delete(smsList);
            } catch (Exception e) {
                e.printStackTrace();
            }
    }
}

File: app/src/main/kotlin/ua/com/todd/smsforwading/MailSenderService.kt
package ua.com.todd.smsforwading

import android.app.IntentService
import android.content.Intent

import org.apache.commons.mail.DefaultAuthenticator
import org.apache.commons.mail.SimpleEmail

import ua.com.todd.baseapp.utils.Utils
import ua.com.todd.smsforwading.data.HelperFactory
import ua.com.todd.smsforwading.managers.PreferenceManager
import kotlin.properties.Delegates

public class MailSenderService : IntentService(javaClass<MailSenderService>().getName()) {

    private var login: String by Delegates.notNull()
    private var pass: String by Delegates.notNull()
    private var host: String by Delegates.notNull()
    private var port: Int by Delegates.notNull()

    override fun onCreate() {
        super.onCreate()
        val app = getApplication() as MyApplication
        with(app.getPreferenceManager<PreferenceManager>()) {
            login = restoreEmail()
            pass = restorePassword()
            host = restoreHost()
            port = restorePort()
        }
    }

    override fun onHandleIntent(intent: Intent) {
        val pref = MyApplication.app().getPreferenceManager<PreferenceManager>()
        val items = HelperFactory.getHelper().getProfileDAO()?.getAllItems()
        if (items!!.isNotEmpty() && pref.isEnabled() && Utils.isNotNull(login, pass, host, port))
            try {
                val smsList : List<Sms> = HelperFactory.getHelper().getSmsDAO()!!.getAllSms()
                for (s in smsList)
                    with(SimpleEmail()) {
                        setHostName(host)
                        setSmtpPort(port)
                        setStartTLSEnabled(true)
                        setAuthenticator(DefaultAuthenticator(login, pass))
                        setFrom(login)
                        setSubject("TestMail")
                        setMsg(s.toString())
                        items.map { addTo(it.mail) }
                        send()
                        HelperFactory.getHelper().getSmsDAO()?.delete(s)
                    }
            } catch (e: Exception) {
                e.printStackTrace()
            }

    }

    fun n(vararg n: String) {
        n.clone()
    }
}
